aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -956,19 +990,36 @@ module aptos_framework::delegation_pool {
     public entry fun set_operator(
         owner: &signer,
         new_operator: address
-    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords {
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords, Beneficiary {
         let pool_address = get_owned_pool_address(signer::address_of(owner));
         // synchronize delegation and stake pools before any user operation
         // ensure the old operator is paid its uncommitted commission rewards
         synchronize_delegation_pool(pool_address);
         stake::set_operator(&retrieve_stake_pool_owner(borrow_global<DelegationPool>(pool_address)), new_operator);
     }
 
+    /// Allows an operator to change its beneficiary.
+    public entry fun set_beneficiary_for_operator(operator: &signer, new_beneficiary: address) acquires Beneficiary {
+        let operator_addr = signer::address_of(operator);
```

Comment:
Storing `delegation_pool::Beneficiary` within the operator account has some advantages:
- decoupling the beneficiary functionality from the delegation pool
- an operator managing multiple pools collects rewards into the same account always
However, what happens to the commission rewards produced under the old beneficiary and not collected yet? At the next `synchronize_delegation_pool` they will be all granted to the new beneficiary.

Reply:
I left a comment on this as well but I think this makes sense and is needed in case the old beneficiary account is compromised. This gives the operator full control in how to distribute the commissions, they can call synchronize first before changing the beneficiary if they need the unpaid commission to go to the old beneficiary first.

Reply:
I added a comment to explain it.

Reply:
n00b Q:
> an operator managing multiple pools collects rewards into the same account always

Is this what we want?

Reply:
> an operator managing multiple pools
This case does not really exist and is not recommended. However, if it does, it is expected that the commissions from multiple pools will go to the same beneficiary.